### PR TITLE
BULK MISC FIXES: Add error counters; Fix null string handlings

### PR DESCRIPTION
**Pull request by Deep1998**
Add error counters for reader and transformer.
Fix null string handlings

### CODE REVIEW

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/constants/MetricCounters.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.constants;
+
+public class MetricCounters {
+
+  // Counters for bulk template.
+  public static final String TRANSFORMER_ERRORS = "transformer_errors";
```
Comment (bharadwaj-aditya) [inline v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/constants/MetricCounters.java:21]: Please add a short description on the types of metrics that will be captured in each.

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/transformer/SourceRowToMutationDoFn.java [hunk 2]
```diff
@@ -73,7 +80,9 @@ public void processElement(ProcessContext c) {
       c.output(mutation);
     } catch (Exception e) {
       // TODO: Add DLQ integration once supported.
-      LOG.error("Unable to transform source row to spanner mutation: {}", e.getMessage());
+      LOG.error("Unable to transform source row to spanner mutation: {} {}", e, e.getMessage());
+      e.printStackTrace();
```
Comment (bharadwaj-aditya) [inline v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/transformer/SourceRowToMutationDoFn.java:84]: please remove printstacktrace

File: v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/avro/GenericRecordTypeConvertor.java [hunk 3]
```diff
@@ -89,6 +89,7 @@ public Map<String, Value> transformChangeEvent(GenericRecord record, String srcT
         result.put(spannerColName, value);
       } catch (NullPointerException e) {
         LOG.info("Unable to transform change event: {}", e.getMessage());
+        e.printStackTrace();
```
Comment (bharadwaj-aditya) [inline v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/avro/GenericRecordTypeConvertor.java:92]: please remove printStackTrace

File: v2/spanner-common/src/test/java/com/google/cloud/teleport/v2/spanner/migrations/avro/AvroToValueMapperTest.java [hunk 4]
```diff
@@ -158,6 +158,29 @@ public void testAvroFieldToDouble_UnsupportedType() {
     AvroToValueMapper.avroFieldToDouble(inputValue, SchemaBuilder.builder().booleanType());
   }
 
+  @Test
+  public void testAvroFieldToString_valid() {
+    String result = AvroToValueMapper.avroFieldToString("Hello");
+    assertEquals("Hello", result);
+
+    result = AvroToValueMapper.avroFieldToString("");
```
Comment (bharadwaj-aditya) [inline v2/spanner-common/src/test/java/com/google/cloud/teleport/v2/spanner/migrations/avro/AvroToValueMapperTest.java:166]: Can you add a test for a null value ?

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/transformer/SourceRowToMutationDoFn.java (hunk 2 above)
Comment (Deep1998) [reply v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/transformer/SourceRowToMutationDoFn.java:84]: But that helps with debugging, thats what helped find the null ptr exception.

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM
